document_multiple_records/static/src/js/document.js

```diff
@@ -70,8 +75,8 @@ var _t = instance.web._t,
                 this.on_attachments_loaded([]);
             }
             else {
-                var dom = [ ['attachment_document_ids.res_model', '=', dataset.model], ['attachment_document_ids.res_id', '=', model_id], ['type', 'in', ['binary', 'url']] ];
-                var ds = new instance.web.DataSetSearch(this, 'ir.attachment', dataset.get_context(), dom);
+                var dom = [ ['related_document', '=', true] ];
```

Comment:
@bwrsandman : Your expression  "var dom = [ ['related_document', '=', [dataset.model, model_id]] ]; " is incorrect, related_document is  a boolean and  [dataset.model, model_id]] is a list.
I just add a filter to list attachments where related_document field is true.